### PR TITLE
[elc] add convenient methods to improve access to aggregations

### LINKED ISSUE #2265: Improve access to aggregations of SearchHits

### PR DESCRIPTION
**Pull request by xhaggi**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] **There is a ticket in the bug tracker for the project in our [issue tracker](https://github.
  com/spring-projects/spring-data-elasticsearch/issues)**. Add the issue number to the _Closes #issue-number_ line below
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Closes #2265

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java [hunk 1]
```diff
@@ -28,35 +30,72 @@
...
 
-		aggregations = new ArrayList<>(aggregationsMap.size());
-		aggregationsMap
+		aggregations = new ArrayList<>(aggregationsAsMap.size());
+		aggregationsAsMap
 				.forEach((name, aggregate) -> aggregations.add(new ElasticsearchAggregation(new Aggregation(name, aggregate))));
+
+		if (aggregations.isEmpty()) {
+			this.aggregationsAsMap = Collections.emptyMap();
+		}
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java:67]:
```suggestion
		this.aggregationsAsMap = aggregationsAsMap;
```

File: src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java [hunk 2]
```diff
@@ -28,35 +30,72 @@
...
+
+		if (aggregationsAsMap == null) {
+			Map<String, ElasticsearchAggregation> newAggregationsAsMap = new HashMap<>(aggregations.size());
+			for (ElasticsearchAggregation aggregation : aggregations) {
+				newAggregationsAsMap.put(aggregation.aggregation().getName(), aggregation);
+			}
+			aggregationsAsMap = Collections.unmodifiableMap(newAggregationsAsMap);
+		}
+
+		return aggregationsAsMap;
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java:88]:
```suggestion
		return Collections.unmodifiableMap(aggregationsAsMap)
```

should be all that's needed

**Review by sothawo — CHANGES_REQUESTED** [review]

File: src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java [hunk 3]
```diff
@@ -28,35 +30,72 @@
...
  * @author Peter-Josef Meisch
+ * @author Sascha Woo
  * @since 4.4
  */
 public class ElasticsearchAggregations implements AggregationsContainer<List<ElasticsearchAggregation>> {
 
 	private final List<ElasticsearchAggregation> aggregations;
+	private Map<String, ElasticsearchAggregation> aggregationsAsMap;
 
 	public ElasticsearchAggregations(List<ElasticsearchAggregation> aggregations) {
```
[the remark this answers is not in the document]
Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchAggregations.java:41]: Ok, then I'll remove it.

**Review by xhaggi — COMMENTED** [review]
I removed the ctor with the list type and adjusted the code according to your comments.

**Review by sothawo — APPROVED** [review]

**Pull request merged**